anap1003/assignment2/Assignment2_Test.java

```diff
@@ -0,0 +1,80 @@
+import assignment2.*;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class Assignment2_Test {
+    
+    private BinaryTree buildTree() {
+        BinaryTree<Integer> tree = new BinaryTree<>();
+
+        tree.addNode(7, "");
+        tree.addNode(3, "L");   //       7
+        tree.addNode(4, "R");   //      / \
+        tree.addNode(2, "LL");  //     3   4
+        tree.addNode(5, "LR");  //    / \   \
+        tree.addNode(8, "RR");  //   2   5   8
+        tree.addNode(1, "LLL"); //  / \
+        tree.addNode(6, "LLR"); // 1   6
+        
+        return tree;
+    }
+    
+    @Test
+    public void testAssignmentExampleQ1() {

```

Comment:
You might be interested to check into junit Parameterized tests. They let you set the parameters such as [{6, [2, 3, 7]}, {2, [3, 7]}, {4, [7]}] and have the same test run over all the parameters.